tools/database/src/memtrie.rs

```diff
@@ -269,6 +272,82 @@ impl<'a, 'b> MemtrieSizeCalculator<'a, 'b> {
             }
         }
 
-        Ok(total_size)
+        Ok(ByteSize::b(total_size))
+    }
+}
+
+#[derive(clap::Parser)]
+pub struct FindBoundaryAccountCommand {
+    #[clap(long, help = "ID of the shard to split")]
+    shard_uid: ShardUId,
+}
+
+impl FindBoundaryAccountCommand {
+    pub fn run(
+        &self,
+        home: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let env_filter = EnvFilterBuilder::from_env().verbose(Some("memtrie")).finish()?;
+        let _subscriber = default_subscriber(env_filter, &Default::default()).global();
+        let mut near_config = nearcore::config::load_config(&home, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+        near_config.config.store.load_memtries_for_tracked_shards = true;
+        let genesis_config = &near_config.genesis.config;
+
+        let db_path =
+            near_config.config.store.path.as_ref().cloned().unwrap_or_else(|| home.join("data"));
+        let rocksdb = Arc::new(RocksDB::open(
+            &db_path,
+            &near_config.config.store,
+            Mode::ReadOnly,
+            Temperature::Hot,
+        )?);
+        let store = near_store::NodeStorage::new(rocksdb).get_hot_store();
+        let chain_store = ChainStore::new(
+            store.clone(),
+            true,
+            near_config.genesis.config.transaction_validity_period,
+        );
+        let final_head = chain_store.final_head()?;
+        let block = chain_store.get_block(&final_head.prev_block_hash)?;
+        let block_hash = *block.hash();
+        println!("Block height: {} block hash: {block_hash}", block.header().height());
```

Comment:
You have a tracing subscriber here, consider `info!`. Similarly for most other prints.

Ideally `stdout` should output just the boundary key so that if needed this tool can be used in scripting contexts easily.